packages/react-magma-dom/src/components/NativeSelect/NativeSelect.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+
+import { inputBaseStyles } from '../InputBase';
+import { InputProps } from '../Input';
+import { InputMessage } from '../Input/InputMessage';
+import { Label } from '../Label';
+import { SelectTriggerButton } from '../Select/SelectTriggerButton';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { useIsInverse } from '../../inverse';
+import { I18nContext } from '../../i18n';
+import { useGenerateId } from '../../utils';
+
+/**
+ * @children required
+ */
+export interface NativeSelectProps extends InputProps {
+  disabled?: boolean;
+  hasError?: boolean;
```

Comment:
You shouldn't have to specify a lot of these props here (helperMessage, isInverse, etc) -- they should already be included when you extend the `InputProps`.   

We also shouldn't need a consumer to provide a `hadError` prop.  We should be able to determine this if by whether the `errorMessage` prop is empty.